packages/agoric-cli/src/lib/rpc.js

```diff
@@ -0,0 +1,265 @@
+/* eslint-disable @jessie.js/no-nested-await */
+// @ts-check
+/* global Buffer, fetch, process */
+
+import { NonNullish } from '@agoric/assert';
+
+/**
+ * @template K, V
+ * @typedef {[key: K, val: V]} Entry<K,V>
+ */
+
+export const networkConfigUrl = agoricNetSubdomain =>
+  `https://${agoricNetSubdomain}.agoric.net/network-config`;
+export const rpcUrl = agoricNetSubdomain =>
+  `https://${agoricNetSubdomain}.rpc.agoric.net:443`;
+
+// Env option instead of CLI option so that CLI args can be validated against it
+// XXX lacking ocaps discipline
+// look into https://github.com/tj/commander.js/blob/master/examples/options-env.js
+export const networkId = process.env.AGORIC_NET || 'local';
+
+/**
+ * @typedef {{ rpcAddrs: string[], chainName: string }} MinimalNetworkConfig
+ */
+
+/**
+ *  @param {string} str
+ * @returns {Promise<MinimalNetworkConfig>}
+ */
+const fromAgoricNet = str => {
+  const [netName, chainName] = str.split(',');
+  if (chainName) {
+    return Promise.resolve({ chainName, rpcAddrs: [rpcUrl(netName)] });
+  }
+  return fetch(networkConfigUrl(netName)).then(res => res.json());
+};
+
+/** @type {MinimalNetworkConfig} */
+export const networkConfig =
+  'AGORIC_NET' in process.env
+    ? await fromAgoricNet(NonNullish(process.env.AGORIC_NET))
+    : { rpcAddrs: ['http://0.0.0.0:26657'], chainName: 'agoric' };
+console.warn('networkConfig', networkConfig);
+/**
+ * Petnames depend on names issued by VBANK.
+ */
+export const vBankPetName = {
+  purse: {
+    anchor: 'AUSD',
+    ist: 'Agoric stable local currency',
+  },
+};
```

Comment:
I'm pretty sure this is dead code, since we're using board IDs for object identity now.